vault/core.go

```diff
@@ -2358,9 +2358,11 @@ func (c *Core) postUnseal(ctx context.Context, ctxCancelFunc context.CancelFunc,
 		for i := 0; i < postUnsealFuncConcurrency; i++ {
 			workerChans[i] = make(chan func())
 			go func(i int) {
-				defer wg.Done()
 				for v := range workerChans[i] {
-					v()
+					func() {
+						defer wg.Done()
```

Comment:
We Add every time we send to a channel, so I agree we need to Done every time we read from it.  Why the extra func/defer though?  Couldn't we just have a loop body of `{ v(); wg.Done() }`?

Reply:
Although alternatively, we could instead Add once per postUnsealFuncConcurrency and Done when the worker exits.